File: bench/sieve/sieve.go
// Sieve of Eratosthenes: tight loops over a large flat array.
package main

import "fmt"

func main() {
	const limit = 4000000

	flags := make([]bool, limit+1)
	for i := 0; i <= limit; i++ {
		flags[i] = true
	}

	count := 0
	for p := 2; p <= limit; p++ {
		if flags[p] {
			count++
			for m := p * p; m <= limit; m += p {
				flags[m] = false
			}
		}
	}

	fmt.Println(count)
}

File: bench/sieve/sieve.lua
-- Sieve of Eratosthenes: tight loops over a large flat array.
local limit = 4000000

local flags = {}
for i = 0, limit do
    flags[i] = true
end

local count = 0
for p = 2, limit do
    if flags[p] then
        count = count + 1
        local m = p * p
        while m <= limit do
            flags[m] = false
            m = m + p
        end
    end
end

print(count)
